### DISCUSSION TITLE
Is there any plan regarding a WYSIWYG editor?

### DISCUSSION THREAD

**Discussion by Fnux** [datenstrom/community#873] [Ask a question] [answered]
Hello yellow-folks,

Thanks for the hard work! I appreciate a lot how yellow was built. I have one question regarding the in-page edit(or), though:

I've read here-and-there in the discussions that there are ideas and/or wishes regarding the editor, but I haven't been able to find clear design decisions regarding the limitations and/or a 'roadmap'. Is there any plan regarding a (simple) WYSIWYG? While not being a blocker, I've seen a fair share of non-technical users being annoyed by 'manual' text-formatting.

What are your thoughts on this? It could indeed be difficult to implement this without introducing too much complexity - I'm pretty sure it was already asked, sorry if I missed it!

Regards from Bern, Switzerland,
Timothée

PS: I believe it could be useful to others if you were to add the answer to [yellow-edit](https://github.com/annaesvensson/yellow-edit)'s README.

**Comment by GiovanniSalmeri** [comment]
Hello Timothée!

I am not a core developer, so I cannot answer on their behalf, but I share my opinion. I think that it would be easy enough to write an extension that leverages one of the several wysiwyg engines (e.g. TinyMCE). But I think that it wouldn't fit very well in the philosophy of Yellow "only files and directories", which I appreciate very much. When I read "files" I understand "simple text files", so no wysiwyg. In my little experience, non-technical users complain for one minute about markdown... and then thank you for having learnt a muuuuuuuch more efficient way for writing and editing texts. (Actually, I do not even use the Edit extension, because I prefer to use my usual text editor, and FTP for uploading files.) Excepted the realm of Desktop Publishing, I haven't yet found a case (math, music...) where simply editing a text file is not faster and easier than using a wysisyg user interface, and I guess that many users love Yellow just because it lets you discover a different and better way of working.

But obviously, if someone nonetheless wants to write a wysisyg extension, no one can stop them 😊

Cheers

Giovanni

**Comment by schulle4u** [comment]
Quite ancient, but for completeness see also datenstrom/community#103 for the previous discussion which lead us to the current editor bar solution.

**Comment by annaesvensson** [comment, marked as answer]
It depends on what you mean.

No, there is no grand plan, roadmap or central steering organisation. 

We organise our work around active contributors. Not just around people who code, but everyone with a touch of motivation/talent/responsibility and a genuine interest in working together. If you are interested in working with us, wonderful, you can make non-code contributions such as discussions about product problems, or you can make code contributions such as pull requests with new product features. I think that active participation is one of the best ways to get things done. See datenstrom/community#622, datenstrom/community#554, datenstrom/community#760.

Yes, we are interested in making the editor simpler to use in the future. 

At the moment the edit extension contains all of the backend and frontend functionality for editing a website in the web browser. If someone wants to make a visual editor that comes with a lot of JavaScript, lovely, we can always break it down into multiple smaller extensions. There have also been ideas that started as separate extensions and then were integrated into the editor. I think anything that makes life easier for people is welcome. See datenstrom/community#130, datenstrom/community#440, datenstrom/community#716.

**Comment by Fnux** [comment]
Hello everyone,

Thanks for detailed replies. I wasn't asking for any 'grand plan' - just what was in the maintainer's heads, if anything! I've got my answers :-)

**Comment by annaesvensson** [comment]
Out of curiosity, is there a good WYSIWYG editor for Markdown? What are the must-have features for non-technical users?

Here's what I remember off the top of my head when this topic was discussed the last time. There are many WYSIWYG editors for web browsers that work on HTML, but what we would need for Datenstrom Yellow is an editor that works on Markdown, because that's what we store in the file system. We found some editors that can export Markdown, e.g. [pen](https://github.com/sofish/pen) and [zenpen](https://github.com/tholman/zenpen). Promising but with technical problems. Some editors converted back and forth between Markdown and HTML with mixed results, some supported only a subset of the Markdown syntax and failed completely when they encountered something they didn't know. Perhaps the situation has improved in the meantime. I wonder if there's an editor that supports live editing with text formatting, but ignores everything else.

**Reply by GiovanniSalmeri** [reply]
Some days ago I found these two projects:

https://github.com/nhn/tui.editor
https://github.com/portive/wysimark

Both seem to be quite complete and actively developed.

I admit that I find rather strange a wysiwyg editor which uses Markdown as the internal format: Markdown is in all respects inferior to HTML, it has only the huge advantage that it can be written and edited very easily by hand. (For many years I have been using only Markdown not only for Internet pages, but for any text i write.) But I recognise that a wysiwyg editor could be useful as an alternative (for example as in Wikipedia, which has both a "source editor" and a "visual editor") .
